lua/packer/log.lua

```diff
@@ -100,18 +100,7 @@ log.new = function(config, standalone)
     local console_lineinfo = vim.fn.fnamemodify(info.short_src, ':t') .. ':' .. info.currentline
     local console_string = string.format('[%-6s%s] %s: %s', nameupper, os.date '%H:%M:%S', console_lineinfo, msg)
 
-    if config.highlights and level_config.hl then
-      vim.cmd(string.format('echohl %s', level_config.hl))
-    end
-
-    local split_console = vim.split(console_string, '\n')
-    for _, v in ipairs(split_console) do
-      vim.cmd(string.format([[echom "[%s] %s"]], config.plugin, vim.fn.escape(v, '"')))
-    end
-
-    if config.highlights and level_config.hl then
-      vim.cmd 'echohl NONE'
-    end
+    vim.notify(string.format([[[%s] %s]], config.plugin, console_string), vim.log.levels[level_config.name:upper()])
```

Comment:
For users of things like `nvim-notify` I think adding a title via the third options is a nice to have i.e. looks much better, but this is a minor aesthetic point.

Reply:
Oh, good point. The only annoying challenge is figuring out when we can remove the leading plugin name in the message, to prevent redundancy...

Reply:
Added a heuristic check for whether `vim.notify` is the fancy version or the built-in.